### DISCUSSION TITLE
Compilation issue while building locally

### DISCUSSION THREAD

**Discussion by pradi0809** [kafbat/kafka-ui#494] [General]
The method below in the class [`KafkaConnectService.java`](https://github.com/kafbat/kafka-ui/blob/main/api/src/main/java/io/kafbat/ui/service/KafkaConnectService.java) fails to compile when built locally . 

```java
public Flux<String> getConnectorNames(KafkaCluster cluster, String connectName) {
    return api(cluster, connectName)
        .flux(client -> client.getConnectors(null))
        // for some reason `getConnectors` method returns the response as a single string
        .collectList().map(e -> e.get(0))
        .map(this::parseConnectorsNamesStringToList)
        .flatMapMany(Flux::fromIterable);
  }
```

This can be rewritten as below:

```java
public Flux<String> getConnectorNames(KafkaCluster cluster, String connectName) {
    return api(cluster, connectName)
        .mono(client -> client.getConnectors(null))
        .flatMapMany(Flux::fromIterable)
        // for some reason `getConnectors` method returns the response as a single string
        .collectList().map(e -> e.get(0))
        .map(this::parseConnectorsNamesStringToList)
        .flatMapMany(Flux::fromIterable);
  }
```

By the way, to make the above code work, another change is required in the class [`RetryingKafkaConnectClient.java`](https://github.com/kafbat/kafka-ui/blob/main/api/src/main/java/io/kafbat/ui/client/RetryingKafkaConnectClient.java) to the below method.

```java
@Override
  public Flux<String> getConnectors(String search) throws WebClientResponseException {
    return withRetryOnConflict(super.getConnectors(search));
  }
```

This needs to be rewritten as below:

```java
  @Override
  public Mono<List<String>> getConnectors(String search) throws WebClientResponseException {
    return withRetryOnConflict(super.getConnectors(search));
  }
```

Let me know if this is a viable option. Thnx!

**Comment by Haarolean** [comment]
Hi, this code is about 1 year old according to git blame, and it perfectly compiles both locally and in CI. If it doesn't for you, please provide exact steps to reproduce.

Replacing this with Mono<List> makes no sense as it levels out the benefits of having a Flux.

**Reply by pradi0809** [reply]
Here is the stack I am using to compile the code from `main` branch.

Corretto-17.0.11.9.1
Apache Maven 3.9.6
Macbook Pro M3 chipset

Here is the command I am issuing as per the documentation:

```sh
./mvnw clean install -Dmaven.test.skip=true -Pprod
```

**Reply by Haarolean** [reply]
can you provide the logs/output?
